### DISCUSSION TITLE
Cache is hit for every single object in collection

### DISCUSSION THREAD

**Discussion by gigerIT** [spatie/laravel-data#667] [Q&A] [answered]
I've just upgraded to v4 and I'm really interested to use the new caching feature (haven't used it in v3).
One thing i immediately noticed that for a collection the cache is accessed three times for every time a data object within a collection is created.

Is this behavior expected? My testing shows that object creation is actually faster without utilizing caching structures, which would be quite disappointing.

```php
Route::get('test/data', function () {
	$agencies = Agency::limit(100)->get();
	return AgencyData::collect($agencies);
});
```

Without cache: (440ms)
<img width="894" alt="image" src="https://github.com/spatie/laravel-data/assets/64487711/facfb64e-fc93-4222-a551-b58a7d6a366d">

With cache: (718ms)
<img width="896" alt="image" src="https://github.com/spatie/laravel-data/assets/64487711/6387f927-c3b7-4251-a7a1-d76abab91f01">
(and so on... cache is hit 300 times for the same object)

**Comment by rubenvanassche** [comment, marked as answer]
Jup that's a bug, fixed in the next release

**Comment by gigerIT** [comment]
The newest update did fix it, thank you very much 👍
